src/Body.php

```diff
@@ -0,0 +1,33 @@
+<?php
+
+namespace Http\Client;
+
+use Psr\Http\Message\StreamInterface;
+
+/**
+ * Allows a more input types for body data
+ *
+ * @author Márk Sági-Kazár <mark.sagikazar@gmail.com>
+ */
+interface Body
+{
+    /**
+     * Returns a set of headers which is needed to correctly send the body
+     *
+     * Note: these headers SHOULD be overwritten if any matching header is passed manually
```

Comment:
maybe: "Note: When creating the HTTP message, the factory SHOULD overwrite matching headers returned by this method with those that are passed manually." ? and could we change that to "factory MUST" ?